collect_app/src/main/java/org/odk/collect/android/formlists/sorting/FormListSortingAdapter.kt

```diff
@@ -0,0 +1,56 @@
+package org.odk.collect.android.formlists.sorting
+
+import android.content.res.ColorStateList
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.core.graphics.drawable.DrawableCompat
+import androidx.recyclerview.widget.RecyclerView
+import org.odk.collect.android.R
+import org.odk.collect.android.databinding.SortItemLayoutBinding
+import org.odk.collect.androidshared.system.ContextUtils.getThemeAttributeValue
+
+class FormListSortingAdapter(
+    private val sortingOptions: List<FormListSortingOption>,
+    private val selectedSortingOrder: Int,
+    private val listener: (position: Int) -> Unit
+) : RecyclerView.Adapter<FormListSortingAdapter.ViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val binding = SortItemLayoutBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        return ViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        with(holder) {
```

Comment:
U used it to avoid repeating `holder.`
Do you like this version better?:
```
        holder.binding.root.setOnClickListener {
            listener.invoke(position)
            selectItem(holder.binding)
        }

        val sortingOption = sortingOptions[position]
        
        holder.binding.title.setText(sortingOption.text)
        holder.binding.icon.setImageResource(sortingOption.icon)
        holder.binding.icon.tag = sortingOption.icon
        holder.binding.icon.setImageDrawable(
            DrawableCompat.wrap(holder.binding.icon.drawable).mutate()
        )

        if (position == selectedSortingOrder) {
            selectItem(holder.binding)
        }
```

Reply:
I do actually find that easier to read. I think I find the double nesting hard to track and I do think I find `with` harder to follow that `apply`/`also`.

Reply:
I don't have any strong preferences so ok we can use that version.